sdb/command.py

```diff
@@ -225,14 +234,145 @@ def __init_subclass__(cls, **kwargs):
 
     def _call(self,
               objs: Iterable[drgn.Object]) -> Optional[Iterable[drgn.Object]]:
-        # pylint: disable=missing-docstring
-        raise NotImplementedError
+        """
+        Optionally implemented by the user.
+
+        Commands that expect to manipulate and deal with any errors
+        of all the objects from the previous stage of the pipeline
+        passed as an iterable, should implement this method.
+
+        Takes precedence over _call_one() - see call() method.
+        """
+        #
+        # We should never get here!
+        #
+        assert self.name is None
+        assert False
+        return objs
+
+    def _call_one(self, obj: drgn.Object) -> Optional[drgn.Object]:
+        """
+        Optionally implemented by the user.
+
```

Comment:
I wonder if a better mechanism for the "call one" would be to have an abstract subclass  of Command (e.g. `SingleInputCommand`) that would do the iteration and print/discard the exception?  And then if you want to use this, you would subclass from SingleInputCommand and override its _call_one method.

Reply:
So I did think about this as it would probably help us with some of the trickiness of the current solution, but then I thought about the scenario of someone trying to create a command that is a `SingleInputCommand, Locator, PrettyPrinter`. Things should work as expected maybe(?) but I couldn't tell if that'd be more tricky (weird inheritance hierarchy) than what we already have. I can create a separate branch maybe an we can compare the two.